Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/RiskLevelTask.kt

```diff
@@ -0,0 +1,166 @@
+package de.rki.coronawarnapp.risk
+
+import android.content.Context
+import com.google.android.gms.nearby.exposurenotification.ExposureSummary
+import de.rki.coronawarnapp.CoronaWarnApplication
+import de.rki.coronawarnapp.exception.ExceptionCategory
+import de.rki.coronawarnapp.exception.RiskLevelCalculationException
+import de.rki.coronawarnapp.exception.reporting.report
+import de.rki.coronawarnapp.nearby.ENFClient
+import de.rki.coronawarnapp.nearby.InternalExposureNotificationClient
+import de.rki.coronawarnapp.risk.RiskLevel.INCREASED_RISK
+import de.rki.coronawarnapp.risk.RiskLevel.LOW_LEVEL_RISK
+import de.rki.coronawarnapp.risk.RiskLevel.NO_CALCULATION_POSSIBLE_TRACING_OFF
+import de.rki.coronawarnapp.risk.RiskLevel.UNDETERMINED
+import de.rki.coronawarnapp.risk.RiskLevel.UNKNOWN_RISK_INITIAL
+import de.rki.coronawarnapp.risk.RiskLevel.UNKNOWN_RISK_OUTDATED_RESULTS
+import de.rki.coronawarnapp.risk.RiskLevel.UNKNOWN_RISK_OUTDATED_RESULTS_MANUAL
+import de.rki.coronawarnapp.storage.LocalData
+import de.rki.coronawarnapp.storage.RiskLevelRepository
+import de.rki.coronawarnapp.task.Task
+import de.rki.coronawarnapp.task.TaskCancellationException
+import de.rki.coronawarnapp.task.TaskFactory
+import de.rki.coronawarnapp.task.common.DefaultProgress
+import de.rki.coronawarnapp.util.ConnectivityHelper
+import de.rki.coronawarnapp.util.ConnectivityHelper.isNetworkEnabled
+import de.rki.coronawarnapp.util.di.AppContext
+import kotlinx.coroutines.channels.ConflatedBroadcastChannel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.asFlow
+import kotlinx.coroutines.flow.first
+import org.joda.time.Duration
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Provider
+
+class RiskLevelTask @Inject constructor(
+    private val riskLevels: RiskLevels,
+    @AppContext private val context: Context,
+    private val enfClient: ENFClient
+) : Task<DefaultProgress, RiskLevelTask.Result> {
+
+    private val internalProgress = ConflatedBroadcastChannel<DefaultProgress>()
+    override val progress: Flow<DefaultProgress> = internalProgress.asFlow()
+
+    private var isCanceled = false
+
+    override suspend fun run(arguments: Task.Arguments): Result {
+        try {
+            Timber.d("Running with arguments=%s", arguments)
+            // If there is no connectivity the transaction will set the last calculated
+            // risk level
+            if (!isNetworkEnabled(context)) {
+                RiskLevelRepository.setLastCalculatedRiskLevelAsCurrent()
+                return Result(UNDETERMINED)
+            }
+            with(riskLevels) {
```

Comment:
What about removing one level of indentation here?

```kotlin

return when {
....
}
.let { Result(it) }
.also { ... }


```

Reply:
I believe it reduces the good overview that you quickly get in the current state by just looking at the code.

Reply:
Agree with @chris-cwa currently it's just nice to read.